### PR TITLE
Feature/install from sql file

### LINKED ISSUE #223: [feature request] Allow dump file for install:from
As a new member to project I might not have access to all the environments yet, and database dump file is easier to get access to than the access required to run `install:from host-config`

But using this method (file) to get the database is rather complex at the moment

Normally I'd do that with `restore:sql-from-file`, but that fails as database does not exist

To create database I would run `drush sql:create`, but that fails because drush does not exist (composer install)

So these two steps above are performed by `install:from` but not with `restore:sql-from-file`

This can be done by running `install` first followed by `restore:sql-from-file`, but install can be really slow.

**Feature request:**

Add a command that merges the functionality of  `install:from host-config` and `restore:sql-from-file`

### PR DESCRIPTION
**Pull request by stmh**
Adds a new command `install:from-sql-file` which will install first the application (skipping a config import), import the sql and then run a reset afterwards.  (Fixes #223)

### CODE REVIEW

File: src/Command/InstallFromSqlFileCommand.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+     * @throws \Phabalicious\Exception\ShellProviderNotFoundException
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        if ($result = parent::execute($input, $output)) {
+            return $result;
+        }
+        $context = $this->getContext();
+        // We are not interested in a config import when using drupal.
+        $context->set(DrushMethod::SKIP_CONFIGURATION_IMPORT, true);
```
Comment (mikran) [inline src/Command/InstallFromSqlFileCommand.php:67]: This comment is very confusing, and the `SKIP_CONFIGURATION_IMPORT` part is not working. Site install is getting exucuted with `--existing-config` and that leads to long waiting time

**Review by mikran — CHANGES_REQUESTED** [review]
SKIP_CONFIGURATION_IMPORT did not work

File: src/Command/InstallFromSqlFileCommand.php (hunk 1 above)
Comment (stmh) [reply src/Command/InstallFromSqlFileCommand.php:67]: hi @mikran 

Are you sure? Can you run the command again with `-v`. In my testing the config import is skipped when running the Drupal install. Please note that as a last step the `reset`-command is executed which will run a config import.

Here's the log from one of my test runs:

Setting up the db:
```
[notice] [96126c] cd /var/www/web && mysql --no-defaults -u root -h mysql --port 3306 -p'admin' -e 'CREATE DATABASE IF NOT EXISTS `controlling`; GRANT ALL PRIVILEGES ON `controlling`.* TO '\''root'\''@'\''%'\''; FLUSH PRIVILEGES;'
```

running drush site-install
```
[notice] [96126c] cd /var/www/web/sites/default && stat /var/www/web/sites/default/settings.php > /dev/null 2>&1
[notice] [96126c] cd /var/www/web/sites/default && stat /var/www/web/../config/sync/core.extension.yml > /dev/null 2>&1
[notice] [96126c] cd /var/www/web/sites/default && grep -q "^\$settings\['config_sync_directory'] = '../config/" settings.php
[notice] [96126c] cd /var/www/web && mkdir -p /var/www/web/sites/default
[notice] [96126c] cd /var/www/web/sites/default && chmod u+w /var/www/web/sites/default
[notice] [96126c] cd /var/www/web/sites/default && chmod u+w /var/www/web/sites/default/settings.php
[notice] [96126c] cd /var/www/web/sites/default && /var/www/vendor/bin/drush site-install -y --sites-subdir=default --account-name=admin --account-pass="admin"  --db-url=mysql://root:admin@mysql/controlling --locale=en minimal
 You are about to:
 * DROP all tables in your 'controlling' database.
 // Do you want to continue?: yes.                                              
 [notice] Starting Drupal installation. This takes a while.
 [notice] Performed install task: install_select_language
 [notice] Performed install task: install_select_profile
 [notice] Performed install task: install_load_profile
 [notice] Performed install task: install_verify_requirements
 [notice] Performed install task: install_verify_database_ready
 [notice] Performed install task: install_base_system
 [notice] Performed install task: install_bootstrap_full
 [notice] Performed install task: install_profile_modules
 [notice] Performed install task: install_profile_themes
 [notice] Performed install task: install_install_profile
 [notice] Performed install task: install_configure_form
 [notice] Performed install task: install_finished
 [success] Installation complete.
                                                                                                                        
 [OK] mbb installed successfully!                                                                                       
                                                                                                                        
```

importing the sql

```
 // Copying dump to `/tmp/mbb.20220204103044.controlling.factorial.io.2022-01-31-22-01-09.sql.gz` ...                   

[notice] [96126c] running command: /usr/bin/scp -P 60918 -o PasswordAuthentication=no controlling.factorial.io.2022-01-31-22-01-09.sql.gz root@controlling.test:/tmp/mbb.20220204103044.controlling.factorial.io.2022-01-31-22-01-09.sql.gz
 // Restoring from `/tmp/mbb.20220204103044.controlling.factorial.io.2022-01-31-22-01-09.sql.gz` ...                    

[notice] Dropping all tables from database ...
[notice] [96126c] cd /var/www/web/sites/default && set -o pipefail
[notice] [96126c] cd /var/www/web/sites/default && mysqldump -u root -h mysql --port 3306 -p'admin' --no-data --single-transaction controlling | grep ^DROP > /tmp/f8b145578021ec8b--drop-tables.sql
[notice] [96126c] cd /var/www/web/sites/default && cat /tmp/f8b145578021ec8b--drop-tables.sql | mysql --no-defaults -u root -h mysql --port 3306 -p'admin' controlling
[notice] [96126c] cd /var/www/web/sites/default && rm /tmp/f8b145578021ec8b--drop-tables.sql
[notice] [96126c] cd /var/www/web/sites/default && set +o pipefail
[notice] Restoring db from /tmp/mbb.20220204103044.controlling.factorial.io.2022-01-31-22-01-09.sql.gz ...
[notice] [96126c] cd /var/www/web/sites/default && gunzip -c /tmp/mbb.20220204103044.controlling.factorial.io.2022-01-31-22-01-09.sql.gz | mysql --no-defaults -u root -h mysql --port 3306 -p'admin' controlling
[notice] [96126c] cd /var/www/web/sites/default && rm /tmp/mbb.20220204103044.controlling.factorial.io.2022-01-31-22-01-09.sql.gz
SQL dump imported successfully
```

running reset task
```
[notice] [96126c] cd /var/www && composer install
Gathering patches for root package.
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Nothing to install or update
Package container-interop/container-interop is abandoned, you should avoid using it. Use psr/container instead.
Package doctrine/reflection is abandoned, you should avoid using it. Use roave/better-reflection instead.
Package drupal/dev-dependencies is abandoned, you should avoid using it. Use drupal/core-dev instead.
Package phpunit/php-token-stream is abandoned, you should avoid using it. No replacement was suggested.
Package sebastian/resource-operations is abandoned, you should avoid using it. No replacement was suggested.
Package swiftmailer/swiftmailer is abandoned, you should avoid using it. Use symfony/mailer instead.
Generating autoload files
85 packages you are using are looking for funding.
Use the `composer fund` command to find out more!
[notice] [96126c] cd /var/www/web/sites/default && /var/www/vendor/bin/drush en -y bi2_deploy
 [notice] Already enabled: bi2_deploy
[notice] [96126c] cd /var/www/web/sites/default && stat /var/www/web/modules_enabled.txt > /dev/null 2>&1
[notice] [96126c] cd /var/www/web/sites/default && stat /var/www/web/modules_disabled.txt > /dev/null 2>&1
[notice] [96126c] cd /var/www/web/sites/default && /var/www/vendor/bin/drush updb -y --no-cache-clear
 [success] No pending updates.
[notice] [96126c] cd /var/www/web/sites/default && /var/www/vendor/bin/drush cr -y
 [success] Cache rebuild complete.
[notice] [96126c] cd /var/www/web/sites/default && /var/www/vendor/bin/drush cset system.site uuid 7383d3ef-e9c7-40d5-8bcb-bacdd7dde2f3 -y
 // Do you want to update uuid key in system.site config?: yes.                 
[notice] [96126c] cd /var/www/web/sites/default && stat /var/www/web/sites/default/settings.php > /dev/null 2>&1
[notice] [96126c] cd /var/www/web/sites/default && stat /var/www/web/../config/sync/core.extension.yml > /dev/null 2>&1
[notice] [96126c] cd /var/www/web/sites/default && grep -q "^\$settings\['config_sync_directory'] = '../config/" settings.php
[notice] [96126c] cd /var/www/web/sites/default && /var/www/vendor/bin/drush cim -y
+------------+--------------------------------------------+-----------+
| Collection | Config                                     | Operation |
+------------+--------------------------------------------+-----------+
|            | rest.resource.bi2_app_search               | Create    |
|            | rest.resource.availability_rest_resource   | Create    |
|            | zmq_notification.settings                  | Create    |
|            | core.extension                             | Update    |
|            | core.entity_form_display.user.user.default | Update    |
|            | user.role.freelancer                       | Update    |
|            | user.role.team                             | Update    |
|            | rest.resource.availability_rest_resource   | Delete    |
+------------+--------------------------------------------+-----------+
 // Import the listed configuration changes?: yes.                              
 [notice] Synchronized extensions: install zmq_notification.
 [notice] Synchronized configuration: delete rest.resource.availability_rest_resource.
 [notice] Synchronized configuration: create rest.resource.bi2_app_search.
 [notice] Synchronized configuration: create rest.resource.availability_rest_resource.
 [notice] Synchronized configuration: create zmq_notification.settings.
 [notice] Synchronized configuration: update core.entity_form_display.user.user.default.
 [notice] Synchronized configuration: update user.role.freelancer.
 [notice] Synchronized configuration: update user.role.team.
 [notice] Finalizing configuration synchronization.
 [success] The configuration was imported successfully.
 ```

I am adding another option to let phab skip the last reset step...

Comment (mikran) [reply src/Command/InstallFromSqlFileCommand.php]: With -v it is showing lots of warnings like 

```
PHP Warning:  Invalid argument supplied for foreach() in /home/mikko/projects/phabalicious/src/Utilities/PasswordManager.php on line 434
PHP Stack trace:
PHP   1. {main}() /home/mikko/projects/phabalicious/bin/phab:0
```

I'll pm the log for you

**Comment by stmh** [comment]
hi @mikran 

please try again, fixed the underlying issue.

**Review by mikran — APPROVED** [review]
Now it works, and it is just what I wanted :+1:

**Pull request merged**